exchanges/bitstamp/bitstamp_test.go

```diff
@@ -871,100 +859,112 @@ func TestWsRequestReconnect(t *testing.T) {
-	if err != nil {
-		t.Fatal(err)
+	c, err := b.GetHistoricCandles(context.Background(), btcusdPair, asset.Spot, kline.OneDay, start, end)
+	assert.NoError(t, err, "GetHistoricCandles should not error")
+	assert.Equal(t, btcusdPair, c.Pair, "Pair should be correct")
+	assert.NotEmpty(t, c, "Candles should not be empty")
+	for _, req := range c.Candles {
+		assert.Positive(t, req.High, "High should be positive")
+		assert.Positive(t, req.Low, "Low should be positive")
+		assert.Positive(t, req.Close, "Close should be positive")
+		assert.Positive(t, req.Open, "Open should be positive")
+		assert.Positive(t, req.Volume, "Volume should be positive")
+		assert.NotEmpty(t, req.Time, "Time should not be empty")
 	}
 }
 
 func TestBitstamp_GetHistoricCandlesExtended(t *testing.T) {
-	pair, err := currency.NewPairFromString("BTCUSD")
-	if err != nil {
-		t.Fatal(err)
-	}
 	start := time.Unix(1546300800, 0)
 	end := time.Unix(1577836799, 0)
 
-	_, err = b.GetHistoricCandlesExtended(context.Background(), pair, asset.Spot, kline.OneDay, start, end)
-	if err != nil {
-		t.Fatal(err)
+	c, err := b.GetHistoricCandlesExtended(context.Background(), btcusdPair, asset.Spot, kline.OneDay, start, end)
+	assert.NoError(t, err, "GetHistoricCandlesExtended should not error")
+	assert.Equal(t, btcusdPair, c.Pair, "Pair should be correct")
+	assert.NotEmpty(t, c, "Candles should not be empty")
+	for _, req := range c.Candles {
+		assert.Positive(t, req.High, "High should be positive")
+		assert.Positive(t, req.Low, "Low should be positive")
+		assert.Positive(t, req.Close, "Close should be positive")
+		assert.Positive(t, req.Open, "Open should be positive")
+		assert.Positive(t, req.Volume, "Volume should be positive")
+		assert.NotEmpty(t, req.Time, "Time should not be empty")
 	}
 }
 
 func TestGetRecentTrades(t *testing.T) {
 	t.Parallel()
+
 	currencyPair, err := currency.NewPairFromString("LTCUSD")
-	if err != nil {
-		t.Fatal(err)
-	}
-	_, err = b.GetRecentTrades(context.Background(), currencyPair, asset.Spot)
-	if err != nil {
-		t.Error(err)
+	assert.NoError(t, err, "NewPairFromString should not error")
+
+	tr, err := b.GetRecentTrades(context.Background(), currencyPair, asset.Spot)
+	assert.NoError(t, err, "GetRecentTrades should not error")
+	assert.NotEmpty(t, tr, "Trades should not be empty")
+	for _, req := range tr {
+		assert.Positive(t, req.Amount, "Amount should be positive")
+		assert.Equal(t, currency.NewPairWithDelimiter("ltc", "usd", ""), req.CurrencyPair, "Pair should be correct")
+		assert.Equal(t, asset.Spot, req.AssetType, "AssetType should be set")
+		assert.NotEmpty(t, req.Timestamp, "Timestamp should not be empty")
+		assert.Positive(t, req.Price, "Price should be positive")
+		assert.NotEmpty(t, req.TID, "TID should not be empty")
 	}
 }
 
 func TestGetHistoricTrades(t *testing.T) {
 	t.Parallel()
+
 	currencyPair, err := currency.NewPairFromString("LTCUSD")
 	if err != nil {
```

Comment:
```suggestion
	if err != nil {
```

assert.NoError(t, err, "NewPairFromString should not error")

Reply:
changed